cmd/argocd/commands/app.go

```diff
@@ -1492,9 +1551,33 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	command.Flags().StringVar(&local, "local", "", "Path to a local directory. When this flag is present no git queries will be made")
 	command.Flags().StringVar(&localRepoRoot, "local-repo-root", "/", "Path to the repository root. Used together with --local allows setting the repository root")
 	command.Flags().StringArrayVar(&infos, "info", []string{}, "A list of key-value pairs during sync process. These infos will be persisted in app.")
+	command.Flags().BoolVar(&diffChangesConfirm, "yes", false, "To be used with --diff-changes for auto confirmation to sync app after a preview of diff against the target and live state before syncing app")
+	command.Flags().BoolVar(&diffChanges, "diff-changes", false, "Preview difference against the target and live state before syncing app and wait for user confirmation")
 	return command
 }
 
+// yesNoPrompt Returns true/false based on user input yes/no, error for invalid input
+func yesNoPrompt(r io.Reader, message string) (bool, error) {
```

Comment:
We do have an utility function for user prompts in `util/cli` package, especially one for Yes/No prompts: https://github.com/argoproj/argo-cd/blob/5676ab5c9bc37cf4d8e60968745f30a3eeb98038/util/cli/cli.go#L116